Blog/templates/Blog/post_list.html

```diff
@@ -2,27 +2,73 @@
 
 {% block content %}
     <h1>Posts</h1>
+    <button class="button secondary" id="dropdown_toggle_1" style="margin-left: 80%">Tags</button>
+    <div class="pos-relative">
+        <div class="bg-grayMouse fg-white"
+             data-role="dropdown"
+             data-toggle-element="#dropdown_toggle_1"
+             style="padding: 1em">
+            <a href="{% url 'posts' %}" class="button">No Tag</a>
+            {% if tags %}
+                {% for tag in tags %}
+                    <a href="{% url 'posts' %}?tag={{ tag }}" class="button">{{ tag }}</a>
+                {% endfor %}
+            {% else %}
+                <p>No Tags available...</p>
+            {% endif %}
+        </div>
+    </div>
 
-    {% if post_list %}
+    {% if object_list %}
         <div class="row">
-            {% for post in post_list %}
-                <div class="card cell-lg-5 offset-lg-1 cell-sd-full">
-                    <div class="card-header">
-                        <h2>{{ post.title }}</h2>
-                    </div>
-                    <div class="card-content p-2">
-                        <p>{{ post.content|truncatechars:240 }}</p>
-                    </div>
-                    <div class="card-footer">
-                        <a class="button secondary" href="{{ post.get_absolute_url }}">Read More</a>
-                    </div>
+        {% for post in object_list %}
+            <div class="card cell-lg-5 offset-lg-1 cell-sd-full">
+                <div class="card-header">
+                    <h2>{{ post.title }}</h2>
+                </div>
+                <div class="card-content p-2">
+                    <p>{{ post.content|truncatechars:240 }}</p>
                 </div>
-            {% endfor %}
+                <div class="card-footer">
+                    <a class="button secondary" href="{{ post.get_absolute_url }}">Read More</a>
+                </div>
+            </div>
+        {% endfor %}
 
-        </div>
     {% else %}
         <p>There are no posts available..</p>
     {% endif %}
+{% if post_list.has_other_pages %}
```

Comment:
The locally inconsistent indentation makes it hard to check the HTML structure. Please indent the code accordingly.